gusto/state.py

```diff
@@ -267,16 +267,22 @@ def setup_diagnostics(self):
 
     def setup_dump(self, pickup=False):
 
-        # setup dump files
         # check for existence of directory so as not to overwrite
         # output files
+        # setup dump files
+        # setup checkpoint files
         self.dumpdir = path.join("results", self.output.dirname)
         outfile = path.join(self.dumpdir, "field_output.pvd")
         if self.mesh.comm.rank == 0 and "pytest" not in self.output.dirname \
            and path.exists(self.dumpdir) and not pickup:
             raise IOError("results directory '%s' already exists" % self.dumpdir)
         self.dumpcount = itertools.count()
         self.dumpfile = File(outfile, project_output=self.output.project_fields, comm=self.mesh.comm)
+        self.chkpts = ["chkpt", "chkptbk"]
+        for name in self.chkpts:
+            chkpt_path = path.join(self.dumpdir, name)
+            chkpt = DumbCheckpoint(chkpt_path, mode=FILE_CREATE)
+            self.chkpts[self.chkpts.index(name)] = chkpt
```

Comment:
You need to remember to close these files manually when you're done.  The garbage collector might do it, but it's safer to do so "by hand".

At the end of the timeloop do something like:
```
for chk in self.chkpts:
    chk.close()
```